src/components/queryBuilder/EditorTypeSwitcher.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { SelectableValue } from '@grafana/data';
+import { RadioButtonGroup, ConfirmModal, InlineFormLabel } from '@grafana/ui';
+import { getQueryOptionsFromSql, getSQLFromQueryOptions } from '../queryBuilder/utils';
+import labels from 'labels';
+import { EditorType, CHQuery, defaultCHBuilderQuery, CHSqlQuery } from 'types/sql';
+import { QueryBuilderOptions } from 'types/queryBuilder';
+import isString from 'lodash/isString';
+
+interface CHEditorTypeSwitcherProps {
+  query: CHQuery;
+  onChange: (query: CHQuery) => void;
+  onRunQuery: () => void;
+}
+
+/**
+ * Component for switching between the SQL and Query Builder editors.
+ */
+export const EditorTypeSwitcher = (props: CHEditorTypeSwitcherProps) => {
+  const { query, onChange } = props;
+  const { label, tooltip, switcher, cannotConvert } = labels.components.EditorTypeSwitcher;
+  let editorType: EditorType =
```

Comment:
Good catch, this was renamed from the QueryTypeSwitcher component in v3. This was one of the few components I left almost untouched since it has some special logic for swapping between the SQL Editor and Query Builder